web-ui/src/app/components/latest-blocks/latest-blocks.component.ts

```diff
@@ -70,9 +72,30 @@ export class LatestBlocksComponent implements OnInit, OnDestroy {
   }
 
   private onBlockRetrieved(response: Block[]) {
+    this.countNewestBlocks(response);
```

Comment:
What I would do is to get the list of new blocks and store them in a mutable variable, something like `this.newBlocks = this.getNewBlocks(this.blocks || [], response)` with a function like `private getNewBlocks(currentBlocks: Block[], latestBlocks: Block[]): Block[]`.

This might give the effect that the initial load will paint all blocks which should be acceptable.